elife/postgresql-appdb.sls

```diff
@@ -20,7 +20,7 @@
 db-perms-to-rds_superuser:
 {% if db_exists and app_user_exists %}
```

Comment:
I'm not sure about how one or the other script is chosen to run. Is `*-legacy.sh` supposed to run every time after the first provisioning (where `-fresh.sh` would run)? Or is it only for particular existing RDS databases? But how can the latter case be detected?

Reply:
yes, it's logic is a little hard (open to suggestions).

when this change was introduced, an existing db would have been detected and the legacy script run. that script shifts permissions to the root user and makes sure the app user can do what they need to do. it worked well.

on new instances that legacy script isn't being run because no db is being detected. previously there was just a short message "no database to wrangle". what it does now with new databases is assign some more permissions to the root user. this is only necessary when using RDS because the pseudo-root user doesn't have permission to deal with new objects in a database, only the ones that were there at the time the permissions were assigned - which was just the empty database. What this script does is assign the pseudo-root user permissions on current objects and _all future objects_ too.

Reply:
hrm - I think I see your problem actually. the legacy script will be run on non-legacy databases, but only once and only if the flag doesn't exist. if both states use the same flag we should be fine, but this logic _is_ getting convoluted.

Reply:
I've tweaked the scripts so both are setting the same flag which reduces the complexity of logic a little.